app/src/main/java/com/android/streetworkapp/ui/profile/Profile.kt

```diff
@@ -258,3 +274,79 @@ fun DisplayUserPicture(user: User?, size: Dp, testTag: String) {
                 .testTag(testTag))
   }
 }
+
+/**
+ * This function displays the friend menu, which contains actions to perform on a friend.
+ *
+ * @param showMenu - The state of the menu (expanded or not).
+ * @param friend - The friend to perform actions on.
+ * @param userViewModel - The view model for the user / the friend.
+ * @param context - The context of the application.
+ */
+@Composable
+fun FriendMenu(
+    showMenu: MutableState<Boolean>,
+    friend: User,
+    userViewModel: UserViewModel,
+    context: Context
+) {
+  DropdownMenu(
+      expanded = showMenu.value,
+      onDismissRequest = { showMenu.value = false },
+      modifier = Modifier.testTag("friendMenu")) {
+        val showConfirmDialog = remember { mutableStateOf(false) }
+        var onConfirm = {}
+        var onDismiss = {}
+
+        CustomDialog(
+            showDialog = showConfirmDialog,
+            dialogType = DialogType.CONFIRM,
+            tag = "RemoveFriend",
+            title = context.getString(R.string.RemoveFriendTitle),
+            verbose = false,
+            onSubmit = { onConfirm() },
+            onDismiss = { onDismiss() },
+            Content = ({ Text(context.getString(R.string.RemoveFriendContent, friend.username)) }))
+
+        DropdownMenuItem(
+            modifier = Modifier.testTag("RemoveFriendMenuItem"),
+            onClick = {
+              val friendUID = friend.uid
+              val friendName = friend.username
+
+              val currentUID = userViewModel.currentUser.value?.uid
+
+              if (currentUID != null) {
+                // Set up the confirm function for the dialog
+                Log.d("Profile", "Setting up callback (confirm) for removing friend")
+                onConfirm = {
+                  Log.d("Profile", "Removined friend $friendName")
+                  userViewModel.removeFriend(currentUID, friendUID)
+                  Toast.makeText(context, "Removed $friendName from friends", Toast.LENGTH_SHORT)
+                      .show()
+                  showMenu.value = false
+                  userViewModel.getFriendsByUid(currentUID) // Refresh friend list
+                }
+
+                onDismiss = { showMenu.value = false }
+
+                showConfirmDialog.value = true
+              } else {
+                Log.d("Profile", "Cannot remove friend - Current user is null")
+                Toast.makeText(context, "Error - could not remove friend", Toast.LENGTH_SHORT)
+                    .show()
+              }
+            },
+            text = {
+              // Display content here
+              Row() {
+                Icon(
+                    painter = painterResource(id = R.drawable.person_remove),
+                    contentDescription = "Remove friend",
+                    modifier = Modifier.size(24.dp),
+                    tint = Color.Red)
+                Text("Remove friend", modifier = Modifier.padding(start = 4.dp), color = Color.Red)
```

Comment:
Personally, I'd put this hard coded text in the `strings.xml` file. In case we want to implement the application in several languages, it's a text we'd like to be able to change easily.

Reply:
I also notice that this is the case for Toast, but I think that here it's more important and that you can leave the hard coded string of Toast - Toast is not part of the primary understanding of the app.

Reply:
Thanks for catching that one ! I did not see it. I will define both of them while I am at it :)